VMBackup/main/freezesnapshotter.py

```diff
@@ -129,8 +129,24 @@ def __init__(self, logger, hutil , freezer, g_fsfreeze_on, para_parser, takeCras
 
             if(para_parser.includedDisks != None and CommonVariables.isAnyDiskExcluded in para_parser.includedDisks.keys()):
                 if (para_parser.includedDisks[CommonVariables.isAnyDiskExcluded] == True and (para_parser.includeLunList == None or para_parser.includeLunList.count == 0)):
-                    self.logger.log('Some disks are excluded from backup and LUN list is not present. Setting the snapshot mode to onlyGuest.')
-                    self.takeSnapshotFrom = CommonVariables.onlyGuest
+                    if( CommonVariables.isAnyDirectDriveDiskIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isAnyDirectDriveDiskIncluded] == True):
+                        errMsg = 'DirectDrive disk is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
+                        self.logger.log(errMsg, True, 'Error')
+                        self.hutil.SetExtErrorCode(ExtensionErrorCodeHelper.ExtensionErrorCodeEnum.FailedInputMismatch)
+                    elif( CommonVariables.isVmgsBlobIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isVmgsBlobIncluded] == True):
+                        errMsg = 'VmgsBlob is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
+                        self.logger.log(errMsg, True, 'Error')
+                        self.hutil.SetExtErrorCode(ExtensionErrorCodeHelper.ExtensionErrorCodeEnum.FailedInputMismatch)
+                    elif( CommonVariables.isAnyWADiskIncluded in para_parser.includedDisks.keys() and para_parser.includedDisks[CommonVariables.isAnyWADiskIncluded] == True):
+                        errMsg = 'WADisk is included, so the host must create the snapshot. IsAnyDiskExcluded is true, but, the included LUN list is empty in the extension input, '\
+                                    'which is not allowed for host DoSnapshot. Thus, failing the extension run.'
+                        self.logger.log(errMsg, True, 'Error')
+                        self.hutil.SetExtErrorCode(ExtensionErrorCodeHelper.ExtensionErrorCodeEnum.FailedInputMismatch)
+                    else:
```

Comment:
These 3 conditions should return different error codes so that they can be differentiated.

Reply:
Correct me if i am wrong 
I think by the error code we can come to know that there was a failure because of Lun list was not passed and Logs we can come to know that there was that particular type of disk while taking the snapshot hence failing it.